docs/dev/commitlog-file-format.md

```diff
@@ -63,3 +63,30 @@ Version 2
         crc             : CRC32 of magic, size
         <entries> * N
 ```
+
+Version 3
+---------
+Version 3 of the format is identical to version 2 except it adds N fields of metadata string pairs to the 
+file header.
+
+```
+
+        Segment file header
+
+        magic           : uint32_t - ('S'<<24) |('C'<< 16) | ('L' << 8) | 'C';
+        version         : uint32_t - same as descriptor
+        id              : uint64_t - same as descriptor
+        metadata_size   : uint32_t - total byte size of metadata, including this size marker.
```

Comment:
How does the replayer distinguish between v2 and v3? Ah `version` is 3?

Reply:
Yes. And both file name version and version in the header must match.